### ISSUE TITLE
Optimization stats reporting

### ISSUE THREAD

**Issue by generall** [qdrant/qdrant#3325]
## Current Behavior

Telemetry of the indexing queue reports multiple instant optimizations of the same segments.

## Steps to Reproduce

1. run qdrant 
2. `docker run --rm -it --network=host  qdrant/bfb:dev ./bfb -b 10 -n 100000  --indexing-threshold 1 --keywords 10 --segments 3`
3. `GET telemetry?details_level=2`

```json
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:53.830467387Z",
                      "end_at": "2024-01-04T20:31:53.830467718Z"
                    },
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:53.830466147Z",
                      "end_at": "2024-01-04T20:31:53.830466545Z"
                    },
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:53.830465470Z",
                      "end_at": "2024-01-04T20:31:53.830465792Z"
                    },
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:53.830464653Z",
                      "end_at": "2024-01-04T20:31:53.830465028Z"
                    },
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:53.830462285Z",
                      "end_at": "2024-01-04T20:31:53.830463963Z"
                    },
                    {
                      "name": "indexing",
                      "segment_ids": [
                        5374283427182239000,
                        16309630277875634000
                      ],
                      "status": "done",
                      "start_at": "2024-01-04T20:31:47.221717322Z",
                      "end_at": "2024-01-04T20:31:53.830454022Z"
                    },
```

## Expected Behavior

Report only those optimizations that actually took place

## Possible Solution

Ignore optimizations that were skipped after scheduling

**Comment by timvisee** [comment]
<https://github.com/qdrant/qdrant/pull/3115> fixes this, because it stops scheduling further optimizations once the configured limit is reached.

**Comment by timvisee** [comment]
<https://github.com/qdrant/qdrant/pull/3364> (part of <https://github.com/qdrant/qdrant/pull/3115>) fixes this issue. It internally checks the optimizations limit each time before scheduling new ones, which prevents this issue.

After it is merged, we want to merge <https://github.com/qdrant/qdrant/pull/3115>. It introduces new optimizer behavior, thus we should confirm the issue remains fixed.

We can close this issue once both PRs are merged and we have confirmed the issue is solved.

**Comment by timvisee** [comment]
It looks like this is fixed in #3364.

If we somehow come across this again, we may reopen this.
